primefaces/src/main/java/org/primefaces/renderkit/CoreRenderer.java

```diff
@@ -878,4 +879,15 @@ protected String getIconOnlyButtonText(String title, String ariaLabel) {
         String text = (title != null) ? title : ariaLabel;
         return (text != null) ? text : "ui-button";
     }
+
+    /**
+     * Logs a WARN log message in ProjectStage == DEVELOPMENT.
+     * @param context the FacesContext
+     * @param message the message to log
+     */
+    protected void logDevelopmentWarning(FacesContext context, String message) {
```

Comment:
Maybe `Supplier<String>` instead of `String`? Saves some code execution in production.

Reply:
will fix